### PR TITLE
Modbus/2

### PR DESCRIPTION
**Pull request by DominikaDusikova**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,15 +54,16 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
-  - IE Databus Configurator V1.7.8
-- Industrial Edge Device (OS) V1.8.0-6
-  - IE Databus V1.7.1
-  - IE Flow Creator V1.10.0-3
-  - IIH Configurator V1.5.0
-  - IIH Registry Service V1.5.0
-  - Modbus TCP Connector V2.0.0-6
-  - Data Service V1.5.0
+- Industrial Edge Management (IEM) V1.5.1-4 / V2.0.1-1
+  - IE Databus Configurator V2.3.1-4
```
Comment (RCAutomatizacion) [inline README.md]: Just "Databus Configurator". Also, latest version is V2.3.2-2
I have tried it and it works with no issues.

File: README.md [hunk 2]
```diff
@@ -54,15 +54,16 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
+- Industrial Edge Virtual Device (OS) ievd-1.14.1-1-b
+  - IE Databus V2.1.0-2
```
Comment (RCAutomatizacion) [inline README.md]: Just "Databus". Also, latest version is V2.3.2-5
I have tried it and it works with no issues.

File: README.md [hunk 3]
```diff
@@ -54,15 +54,16 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
-- Industrial Edge Device (OS) V1.8.0-6
-  - IE Databus V1.7.1
-  - IE Flow Creator V1.10.0-3
-  - IIH Configurator V1.5.0
-  - IIH Registry Service V1.5.0
-  - Modbus TCP Connector V2.0.0-6
-  - Data Service V1.5.0
+- Industrial Edge Management (IEM) V1.5.1-4 / V2.0.1-1
+  - IE Databus Configurator V2.3.1-4
+- Industrial Edge Virtual Device (OS) ievd-1.14.1-1-b
```
Comment (RCAutomatizacion) [inline README.md]: Maybe here you can put the ievd-1.16.1-1-a. I have tried it with that version with no issues.

File: README.md [hunk 4]
```diff
@@ -54,15 +54,16 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
+  - IE Databus V2.1.0-2
+  - IE Flow Creator V1.16.0-2
+  - IIH Configurator V1.9.0-4
```
Comment (RCAutomatizacion) [inline README.md]: "Common Configurator"

File: README.md [hunk 5]
```diff
@@ -54,15 +54,16 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
-  - IE Flow Creator V1.10.0-3
-  - IIH Configurator V1.5.0
-  - IIH Registry Service V1.5.0
-  - Modbus TCP Connector V2.0.0-6
-  - Data Service V1.5.0
+- Industrial Edge Management (IEM) V1.5.1-4 / V2.0.1-1
+  - IE Databus Configurator V2.3.1-4
+- Industrial Edge Virtual Device (OS) ievd-1.14.1-1-b
+  - IE Databus V2.1.0-2
+  - IE Flow Creator V1.16.0-2
```
Comment (RCAutomatizacion) [inline README.md]: Just "Flow Creator V1.16.0-2"

File: README.md [hunk 6]
```diff
@@ -54,15 +54,16 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
+  - IIH Configurator V1.9.0-4
+  - IIH Registry Service V1.9.0.0
+  - Modbus TCP Connector V3.0.1
+  - IIH Essentials V1.9.0
+  - Databus Gateway V1.9.0.0
 - Modbus TCP Server
```
Comment (RCAutomatizacion) [inline README.md]: Did you use "ModRSsim2 (Modbus TCP Server Simulator)" to simulate the Modbus server? I would be good to put it there.

File: docs/Installation.md [hunk 7]
```diff
@@ -60,17 +60,11 @@ With the IIH Configurator, you can configure several connectors and publish the
...
 
-To configure the Modbus TCP Connector, proceed as following:
 
-- go to the tab 'Get data' and select tab 'Databus connectors'
-- select the Modbus TCP Connector
-- switch to tab 'Tags'
-- choose 'Add data source'
-- configure the PLC accordingly and save
 
 **Zero based addressing**: is enabled by default; the configured address should start with an index of zero so that both PLC register addresses and configurator indexes will start from zero.
```
Comment (RCAutomatizacion) [inline docs/Installation.md]: "Zero based addressing" is disabled by default in V3

"Change word order" is not an option anymore with V3

"Use single write" is disabled by default in V3

File: docs/Installation.md [hunk 8]
```diff
@@ -82,20 +76,33 @@ To configure the Modbus TCP Connector, proceed as following:
 05H, 06H, 15H, and 16H are used for writing into the PLC. Otherwise only function codes 15H and 16H are used.
 
 Please look up the operating manual of the app for detailed information.
+In Commoc configurator choose Get Data - Connector configurator
```
Comment (RCAutomatizacion) [inline docs/Installation.md]: "Common Configurator"

File: docs/Installation.md [hunk 9]
```diff
@@ -82,20 +76,33 @@ To configure the Modbus TCP Connector, proceed as following:
...
 Please look up the operating manual of the app for detailed information.
+In Commoc configurator choose Get Data - Connector configurator 
 
-![configuration1](/docs/graphics/Configuration1.png)
+![configuration1](/docs/graphics/connectors.png)
 
-- under column 'Actions' of the newly created PLC, choose 'Add tag'
+To configure the Modbus TCP Connector, proceed as following:
+
+- go to the tab 'Get data' and select tab 'Databus connectors'
```
Comment (RCAutomatizacion) [inline docs/Installation.md]: Delete "...and select tab 'Databus connectors'"

File: docs/Installation.md [hunk 10]
```diff
@@ -82,20 +76,33 @@ To configure the Modbus TCP Connector, proceed as following:
...
+- select the Modbus TCP Connector
+- switch to tab 'Tags'
+- choose 'Add data source'
+- configure the PLC accordingly and save
+ ![configuration2](/docs/graphics/modbusset.png)
+ 
+  under column 'Actions' of the newly created PLC, choose 'Add tag'
```
Comment (RCAutomatizacion) [inline docs/Installation.md]: Add the bullet point here

File: docs/Installation.md [hunk 11]
```diff
@@ -82,20 +76,33 @@ To configure the Modbus TCP Connector, proceed as following:
...
+  
+- ![configuration2](/docs/graphics/tagimport.png)
```
Comment (RCAutomatizacion) [inline docs/Installation.md]: remove the bullet point here

File: docs/Installation.md [hunk 12]
```diff
@@ -82,20 +76,33 @@ To configure the Modbus TCP Connector, proceed as following:
...
+  
 - configure the tags accordingly and save
 
 ![configuration2](/docs/graphics/Configuration2.png)
```
Comment (RCAutomatizacion) [inline docs/Installation.md]: this pic corresponds to v2, please change

File: docs/Usage.md [hunk 13]
```diff
@@ -67,8 +67,7 @@ If some data is written on tag ***var_dint*** with ID 103, the output looks like
```
Comment (RCAutomatizacion) [inline docs/Usage.md:59]: Just put "Flow Creator".

The tags "var_int" and "var_dint" are not the ones you created on the "Installation" page. You have "tag" and "tag2".

File: docs/Usage.md [hunk 14]
```diff
@@ -67,8 +67,7 @@ If some data is written on tag ***var_dint*** with ID 103, the output looks like
 
 ![read_2](/docs/graphics/Read_2.png)
 
-## Use Data Service
-
+## Use IIH Essentials
 The app Data Service collects the data out of different connectors and stores it for a defined time period. This is a prerequisite for other apps like Performance Insight.
```
Comment (RCAutomatizacion) [inline docs/Usage.md]: "IIH Essentials" instead of "Data Service"

File: docs/Usage.md [hunk 15]
```diff
@@ -83,6 +82,14 @@ To activate the data transfer from the Modbus TCP Connector, proceed as followin
```
Comment (RCAutomatizacion) [inline docs/Usage.md]: Please update this pic.

File: docs/Usage.md [hunk 16]
```diff
@@ -83,6 +82,14 @@ To activate the data transfer from the Modbus TCP Connector, proceed as followin
 
 - go to tab 'Assets & Connectivity' and add the variables, that were configured within the Modbus TCP Connector
 
-![DataServiceAdapter](/docs/graphics/DataService_Add.png)
+- ## Use Store Data in Common Configurator
```
Comment (RCAutomatizacion) [inline docs/Usage.md]: From this part, it should be clarified that, to use "Store Data" option, IIH Essentials needs to be integrated into Common Configurator. Therefore, previous step "Use IIH Essentials" is not necessary.

File: docs/Usage.md [hunk 17]
```diff
@@ -83,6 +82,14 @@ To activate the data transfer from the Modbus TCP Connector, proceed as followin
...
 
-![DataServiceAdapter](/docs/graphics/DataService_Add.png)
+- ## Use Store Data in Common Configurator
+
+- open Define Data in Common Configurator
+- choose Modbus TCP Connector
+- select Deploy
+![DataServiceAdapter](/docs/graphics/Definedata.png)
+- open Store Data in Common Configurator
+- create new Asset
```
Comment (RCAutomatizacion) [inline docs/Usage.md]: Delete this step.

File: docs/Usage.md [hunk 18]
```diff
@@ -83,6 +82,14 @@ To activate the data transfer from the Modbus TCP Connector, proceed as followin
 
 - go to tab 'Assets & Connectivity' and add the variables, that were configured within the Modbus TCP Connector
 
-![DataServiceAdapter](/docs/graphics/DataService_Add.png)
+- ## Use Store Data in Common Configurator
+
+- open Define Data in Common Configurator
+- choose Modbus TCP Connector
+- select Deploy
```
Comment (RCAutomatizacion) [inline docs/Usage.md]: Maybe here you can put "Add asset, add tags to the asset and deploy"

**Comment by EvaKaul** [comment]
Hello @DominikaDusikova, 
please implement Rafaels comments, which are all valid points. 
In addition (just a formal point), I would add a border to all screenshots (like it is suggested in the [template repository](https://github.com/industrial-edge/how-to-template/blob/main/docs/Installation.md#build-application))

File: README.md [hunk 19]
```diff
@@ -48,31 +48,32 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
-  - Modbus TCP Connector V2.0.0-6
-  - Data Service V1.5.0
-- Modbus TCP Server
+- Industrial Edge Management (IEM) V1.5.1-4 / V2.0.1-1
+  - Databus Configurator V2.3.2-2
+  - Industrial Edge Virtual Device (OS) ievd-1.16.1-1-a
+  - Databus V2.3.2-5
+  - Flow Creator V1.16.0-2
+  - Common Configurator V1.14.0
+  - IIH Registry Service V1.9.0.0
```
Comment (EvaKaul) [inline README.md]: V1.9.0

File: README.md [hunk 20]
```diff
@@ -48,31 +48,32 @@ Here we configure a connection between a Modbus server and client. The IED acts
...
+  - Modbus TCP Connector V3.0.1
+  - IIH Essentials V1.9.0
+  - Databus Gateway V1.9.0.0
```
Comment (EvaKaul) [inline README.md]: V1.9.0

File: README.md
Comment (EvaKaul) [inline README.md:1]: Please check if the supported PLC variants need to be mentioned or if the connector supports all kinds of PLCs meanwhile (at least in the docu I could not find any hints regarding limitations)

File: docs/Installation.md
Comment (EvaKaul) [inline docs/Installation.md:1]: Replace all "IIH Configurator" by "Common Configurator"

File: docs/Installation.md [hunk 21]
```diff
@@ -60,42 +60,46 @@ With the IIH Configurator, you can configure several connectors and publish the
...
-**Change bit order**: is disabled by default; here the standart bit counting method 16 LSB - 1 MSB is used. Otherwise the bit counting method 0 LSB - 15 MSB is used.
+"Zero based addressing" is disabled by default in V3
+"Change word order" is not an option anymore with V3
+"Use single write" is disabled by default in V3
 
-**Use single write**: is enabled by default; the function codes
 05H, 06H, 15H, and 16H are used for writing into the PLC. Otherwise only function codes 15H and 16H are used.
 
 Please look up the operating manual of the app for detailed information.
+In Common configurator choose Get Data - Connector configurator
```
Comment (EvaKaul) [inline docs/Installation.md]: Please insert a line break before this sentence.

File: docs/Installation.md [hunk 22]
```diff
@@ -60,42 +60,46 @@ With the IIH Configurator, you can configure several connectors and publish the
...
+- go to the tab 'Get data'
+- select the Modbus TCP Connector
+- switch to tab 'Tags'
+- choose 'Add data source'
+- configure the PLC accordingly and save
+ ![configuration2](/docs/graphics/modbusset.png)
+ 
 - under column 'Actions' of the newly created PLC, choose 'Add tag'
+  
+- ![configuration2](/docs/graphics/tagimport.png)
```
Comment (EvaKaul) [inline docs/Installation.md]: remove the bullet point for the screenshot

File: docs/Usage.md
Comment (EvaKaul) [inline docs/Usage.md:1]: replace "Data Service" by "IIH Essentials" (also in the link at the top of the page!)

File: docs/Usage.md [hunk 23]
```diff
@@ -83,6 +75,11 @@ To activate the data transfer from the Modbus TCP Connector, proceed as followin
 
 - go to tab 'Assets & Connectivity' and add the variables, that were configured within the Modbus TCP Connector
 
-![DataServiceAdapter](/docs/graphics/DataService_Add.png)
+- ## Use Store Data in Common Configurator
```
Comment (EvaKaul) [inline docs/Usage.md]: remove the bullet points here

File: docs/Usage.md [hunk 24]
```diff
@@ -83,6 +75,11 @@ To activate the data transfer from the Modbus TCP Connector, proceed as followin
 
 - go to tab 'Assets & Connectivity' and add the variables, that were configured within the Modbus TCP Connector
 
-![DataServiceAdapter](/docs/graphics/DataService_Add.png)
+- ## Use Store Data in Common Configurator
+- (It´s necessary to have IIH Essentials installed)
+- open Define Data in Common Configurator
+- choose Modbus TCP Connector
+- Add asset, add tags to the asset and deploy
+
```
Comment (EvaKaul) [inline docs/Usage.md:75]: Isn't it also necessary to select the "Storage" attribute here, if you want to store the data in IIH Essentials? Maybe an additional screenshot would be helpful.

File: docs/Usage.md
Comment (EvaKaul) [inline docs/Usage.md:1]: Please add borders to all your new screenshots to have an unique appearance here.

**Comment by EvaKaul** [comment]
Hi @DominikaDusikova, I just read over your changes an found some additional points (see comments above). Please implement these changes as well - thank you 😀

**Comment by EvaKaul** [comment]
I updated the Usage.md (removed chapter listing supported PLCs, since any Modbus PLC is supported now).

I updated the Installation.md (fixed "IE Databus" expressions, fixed link in table of content, updated parameter description for adding a data source)

@DominikaDusikova open points are:
- graphic [Configuration2.png](https://github.com/industrial-edge/modbus-tcp-connector-getting-started/blob/Modbus/2/docs/graphics/Configuration2.png) (in Installation.md) -> the used tags do not fit to the others; also the "publish" option is missing in the screenshot
- graphics[ definedata.png ](https://github.com/industrial-edge/modbus-tcp-connector-getting-started/blob/Modbus/2/docs/graphics/definedata.png) and [definedata.png.jpg](https://github.com/industrial-edge/modbus-tcp-connector-getting-started/blob/Modbus/2/docs/graphics/definedata.png.jpg) are not used anywhere; was this supposed?

**Review by EvaKaul — APPROVED** [review]
Checked and adapted everything.
Ready to merge.

**Pull request merged**